zipper/protocols/irondb/irondb_helpers.go

```diff
@@ -0,0 +1,125 @@
+package irondb
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+)
+
+// graphiteExprListToIronDBTagQuery - converts list of Graphite Tag expressions to IronDB Tag query
+// e.g. ["name=~cpu\..*", "tag1!=value1"] => and(__name:/cpu\..*/,not(tag1:value1))
+func graphiteExprListToIronDBTagQuery(exprList []string) string {
+	//r := make([]string, len(exprList))
+	var r []string
+
+	for _, expr := range exprList {
+		// remove escaped quotes
+		expr = strings.ReplaceAll(expr, "\"", "")
+		eq_idx := strings.Index(expr, "=")
+		if eq_idx > 0 {
+			tag_name := expr[:eq_idx]
+			ne_idx := strings.Index(expr, "!")
+			if ne_idx != -1 {
+				tag_name = expr[:ne_idx]
+			}
+			if tag_name == "name" {
+				tag_name = "__name"
+			}
+			if ne_idx == -1 {
+				// Handle = and =~
+				if expr[eq_idx+1] == '~' {
+					// op = "=~"
+					// tag_value = expr[eq_idx+2:]
+					r = append(r, fmt.Sprintf("and(%s:/%s/)", tag_name, expr[eq_idx+2:]))
+				} else {
+					// op = "="
+					// tag_value = expr[eq_idx+1:]
+					r = append(r, fmt.Sprintf("and(%s:%s)", tag_name, expr[eq_idx+1:]))
+				}
+			} else {
+				// Handle != and !=~
+				if expr[eq_idx+1] == '~' {
+					// op = "!~"
+					// tag_value = expr[eq_idx+2:]
+					r = append(r, fmt.Sprintf("not(%s:/%s/)", tag_name, expr[eq_idx+2:]))
+				} else {
+					// op = "!="
+					// tag_value = expr[idx+1:]
+					r = append(r, fmt.Sprintf("not(%s:%s)", tag_name, expr[eq_idx+1:]))
+				}
+			}
+		}
+	}
+	if len(r) > 0 {
```

Comment:
len(r) is always >= 0 so it would be better to do something like:
```
switch len(r) {
   case 0:
       return ""
   case 1:
       return r[0]
   default:
       return fmt.Sprintf...
```

Reply:
Good point! Done